src/shared/codelens/codeLensUtils.ts

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { LambdaHandlerCandidate } from '../lambdaHandlerSearch'
+import {
+    SamCliProcessInvoker,
+    SamCliTaskInvoker
+} from '../sam/cli/samCliInvoker'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+const log = (channel: vscode.OutputChannel, level: LogLevel, msg: string, err?: Error): void => {
+    channel.show()
+    const _msg = `[${level}]: ${msg}`
+    if (err) {
+        // TODO: Define standard/strategy. Sometimes err.message is part of msg. Probably shouldn't be.
+        channel.appendLine(`${msg}: ${ err.message}`)
+        console[level](new Date().toISOString(), _msg, err.stack)
+    } else {
+        channel.appendLine(msg)
+        console[level](new Date().toISOString(), _msg)
+    }
+}
+
+export enum OutputChannelName {
+    ToolKit = 'ToolKit', // localize('AWS.channel.aws.toolkit', 'AWS Toolkit')
+    Lambda = 'Lambda'
+}
+const outputChannels: {[channelName: string]: vscode.OutputChannel} = {
+    [OutputChannelName.ToolKit.toString()]: vscode.window.createOutputChannel(
+        localize('AWS.channel.aws.toolkit', 'AWS Toolkit')
+    ),
+    [OutputChannelName.Lambda.toString()]: vscode.window.createOutputChannel('AWS Lambda')
+}
+
+export const getOutputChannel = (name: OutputChannelName) => outputChannels[name.toString()]
```

Comment:
Functions should be exported as functions rather than consts. This applies throughout this changeset. For example:

```
export async function getLambdaHandlerCandidates({uri}: {uri: vscode.Uri}): Promise<LambdaHandlerCandidate[]>
```

Reply:
"should?" Any reasoning behind that? 

I'd rather not waste much time changing the stuff in the "Temporary logging utilities to be replaced by upcoming logger implementation" section but would like to chat about the use of `debug`, 'info`, `warn`, `error`. 

- What patterns should we use for logging to file vs to output channel(s)? 
- Does it make sense to log everything that we put in the output channel(s) to the file?
- Should adjusting the log level affect what goes to the output channel(s)

Reply:
> I'd rather not waste much time changing the stuff in the "Temporary logging utilities to be replaced by upcoming logger implementation" section.

Sure, that's fair.

> "should?" Any reasoning behind that?

AFAIK there is no functional difference between the two. This feedback is purely for consistency with the rest of our codebase (personally I find the `function` style more readable/concise, but that's subjective).